### ISSUE TITLE
Support for pulling VPN PSK value from Secrets Manager

### ISSUE THREAD

**Issue by in-1911** [terraform-ibm-modules/terraform-ibm-landing-zone#525]
### Description

When a VPN gateway is configured for VPN Landing Zone, the pre-shared key (PSK) value is stored in clear text in override.json.
If a dummy value is used in configuration and the real value is updated outside terraform, subsequent deployments not only  override the correct PSK value but also reveal it in the terraform logs as previous value being changed (I believe it's not marked as sensitive in the provider).

A way to address the issue with storing PSK in configuration would be to add an option to specify a secret name / CRN to pull during provisioning of the VPN gateway resource.
It may not be necessary to actually generate the PSK during provisioning as this may require specific crypto material characteristics.

To indicate a use of an SM secret a new attribute may need to be added to the connection object of vpn_gateway variable. Alternatively, a special prefix in the original preshared_key attribute (e.g. `@secrets-manager:`) may indicate a secret name to be used. 

### New or affected modules

[vpn.tf](/terraform-ibm-modules/terraform-ibm-landing-zone/blob/main/vpn.tf) - get value from an SM secret if requested via VPN connection attributes.
[variables.tf](/terraform-ibm-modules/terraform-ibm-landing-zone/blob/main/variables.tf) - define the SM reference notation in VPN connection attribute of vpn_gateways variable (if additional attribute is used).
[dynamic_values/vpn.tf](/terraform-ibm-modules/terraform-ibm-landing-zone/blob/main/dynamic_values/vpn.tf) - define the SM reference notation in VPN connection attribute of vpn_gateways variable (if additional attribute is used).

---

**Comment by kierramarie** [comment]
We are addressing this issue and after a deep dive with the team have come to some decisions.  For right now, we have verified that if you create a landing zone without the override.json (in other words, with all the defaults) then manually go in and create the connections in the VPN, when applying changes it will not remove the connections. As for the future, we will be deprecating the VPN connections, and they will need to be added manually. Since the override.json is not sensitive, as you mentioned, it is best that the pre-shared key not be listed there. We came to the conclusion that due to the complexity of multiple connections it would be difficult to make the change to use a secrets manager, so we won't be doing that at this time.
